src/objective-c/GRPCClient/GRPCCall.h

```diff
@@ -155,6 +155,19 @@ typedef NS_ENUM(NSUInteger, GRPCErrorCode) {
 };
 
 /**
+ * Flags for options of a gRPC call
+ *
+ */
+typedef NS_OPTIONS(NSUInteger, GRPCCallFlags) {
+  /** Signal that the call is idempotent */
+  GRPCFlagIdempotentRequest = 0x00000010,
+  /** Signal that the call should not return UNAVAILABLE before it has started */
+  GRPCFlagIgnoreConnectivity = 0x00000020,
+  /** Signal that the call is cacheable. GRPC is free to use GET verb */
+  GRPCFlagCacheableRequest = 0x00000040,
```

Comment:
> when it is set "idempotent" is automatically considered set

Is it?

> there might be other flags in the future that are orthogonal to these two flags

We won't need to expose those future flags to the user as part of the same `int`, so that shouldn't be a problem. Think of it this other way:

This is essentially a way to tell the library whether to use `POST`, `PUT`, or `GET` as an HTTP verb for this method. If we were exposing it that way, as a "verb" parameter for which the user can provide one of those values, we would never consider adding other stuff like "ignore connectivity" to the same parameter. It would be unnatural and unnecessary (nothing forces you to cram two different things in the same parameter). And we'd never think of exposing the verbs as a bitmask, rather as an enum: If in the future gRPC wanted to use `DELETE` too, you'd end up with a comment saying "only 1 bit of this mask can be set".

(That's in general what I meant with designing the feature from the user perspective, instead of just exposing what the C library does. The C layer has language constraints that we don't have, and we have requirements of usability and intuitiveness that the C layer doesn't have.)

Another possible design with HTTP verbs would be letting the user set the `@":method"` `requestHeader` of a call: If you started with a design like that, you'd never end up with a bitmask to represent the verb.

Now I'm not saying at all we should expose to the user that this corresponds to an HTTP verb. The point is that what we want to do is attach a three-valued property to a gRPC method. Exposing that as a four-valued property of every call leaves the user prone to mistakes.